File: Easy/Power of Thor/powerOfThor.js
let inputs = readline().split(' ');
let lightX = parseInt(inputs[0]); // the X position of the light of power
let lightY = parseInt(inputs[1]); // the Y position of the light of power
var initialTx = parseInt(inputs[2]); // Thor's starting X position
var initialTy = parseInt(inputs[3]); // Thor's starting Y position

// game loop
while (true) {
    const remainingTurns = parseInt(readline()); // The remaining amount of turns Thor can move. Do not remove this line.

    let directionX = "";
    let directionY = "";

    if(lightY < initialTy){
        directionY ="N";
        initialTy--;
    }else if(lightY > initialTy){
        directionY ="S";
        initialTy++;
    }

    if(lightX < initialTx){
        directionX = "W";
        initialTx--;
    }else if(lightX > initialTx){
        directionX = "E";
        initialTx++;
    }

    // A single line providing the move to be made: N NE E SE S SW W or NW
    console.log(directionY+directionX);
}

File: Easy/Power of Thor/powerOfThor.java
import java.util.*;
import java.io.*;
import java.math.*;

class Player {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int lightX = in.nextInt(); // the X position of the light of power
        int lightY = in.nextInt(); // the Y position of the light of power
        int initialTX = in.nextInt(); // Thor's starting X position
        int initialTY = in.nextInt(); // Thor's starting Y position
        
        int x = initialTX;
        int y = initialTY;

        // game loop
        while (true) {
            int remainingTurns = in.nextInt(); // The remaining amount of turns Thor can move. Do not remove this line.
            
            String move = "";
            
            if (y > lightY) {
                move += "N"; y--;
            }
            else if (y < lightY) {
                move += "S"; y++;
            }
            
            if (x > lightX) {
                move += "W"; x--;
            }
            else if (x < lightX) {
                move += "E"; x++;
            }

            // A single line providing the move to be made:
            // N NE E SE S SW W or NW
            System.out.println(move); 
            
        }
    }
}